kmk/common/internal_state.py

```diff
@@ -182,18 +194,21 @@ def kmk_reducer(state=None, action=None, logger=None):
         state.matrix = action.matrix
         state.keys_pressed |= pressed
         state.keys_pressed -= released
-        state.hid_pending = True
+        if state.leader_mode > 1:  # Active in either Mode
```

Comment:
Basically every check of keys should disable HID when leader mode is active. It has been set in all of these places to ensure that control remains with the leader function while that is active, otherwise ignores it.